src/grpc/keygen/result/gg20.rs

```diff
@@ -6,53 +6,56 @@
 
 use tofn::{gg20::keygen::SecretKeyShare, sdk::api::serialize};
 
-use super::{
-    proto::{self},
-    types::{BytesVec, KeygenInitSanitized, TofnKeygenOutput, TofndKeygenOutput},
-    Gg20Service,
+use crate::{
+    grpc::{
+        keygen::types::{
+            common::{BytesVec, KeygenInitSanitized, KeygenOutput},
+            gg20,
+        },
+        service::Service,
+        types::gg20::PartyInfo,
+    },
+    kv_manager::types::KeyReservation,
+    proto,
 };
-use crate::{gg20::types::PartyInfo, kv_manager::types::KeyReservation};
 
 // tonic cruft
-use tokio::sync::{
-    mpsc,
-    oneshot::{self, Receiver},
-};
+use tokio::sync::mpsc;
 use tonic::Status;
 
 // error handling
 use crate::TofndResult;
 use anyhow::anyhow;
 
-impl Gg20Service {
+fn to_gg20_keygen_outputs(outs: Vec<KeygenOutput>) -> TofndResult<Vec<gg20::TofnKeygenOutput>> {
+    let mut gg20_outs = Vec::with_capacity(outs.len());
+    for out in outs {
+        match out {
+            KeygenOutput::Gg20(gg20_out) => gg20_outs.push(gg20_out?),
+            KeygenOutput::Multisig(_) => return Err(anyhow!("Wrong Keygen type. Expecting Gg20")),
+        }
```

Comment:
I'm not convinced you should need to type-switch like this.  Should be able to do the abstract stuff with traits and/or generics.